### PR TITLE
Apply clippy recommendations to fix master

### PR DESCRIPTION
**Pull request by thanethomson**
`master`'s [currently broken](https://github.com/informalsystems/tendermint-rs/runs/2557422045) due to failing clippy lints, and this attempts to fix it.

* [ ] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG.md

### CODE REVIEW

**Review by xla — APPROVED** [review]
💨 👺 🕴 👼

**Comment by greg-szabo** [comment]
This eludes me right now: the for loop had a `continue` which was turned into the `.for_each` loop's `return`. Does that have the same effect?

File: testgen/src/tester.rs [hunk 1]
```diff
@@ -451,27 +451,25 @@ impl Tester {
...
+                        // ignore path components starting with '_'
+                        if starts_with_underscore(&entry) {
+                            return;
+                        }
+                        if let Ok(kind) = entry.file_type() {
+                            let path = format!("{}", entry.path().display());
+                            let rel_path = self.env().unwrap().rel_path(&path).unwrap();
+                            if kind.is_file() || kind.is_symlink() {
+                                if !rel_path.ends_with(".json") {
+                                    return;
```
[the remark this answers is not in the document]
Comment (greg-szabo) [reply testgen/src/tester.rs:464]: I checked the docs for `for_each` to refresh my knowledge. `for_each` doesn't support `break` or `continue` according to the docs. The `return` here, I'm guessing, will exit the loop.

But the previous code continued to the next item in `paths`. I'm not sure if these will have the same results...

[the remark this answers is not in the document]
Comment (xla) [reply testgen/src/tester.rs:464]: It does, as the for_each closure is invoked for every item.

[the remark this answers is not in the document]
Comment (greg-szabo) [reply testgen/src/tester.rs:464]: Thanks for confirming! My Rust knowledge is a bit Rusty.

**Pull request merged**